src/crypto/crypto_context.cc

```diff
@@ -1308,10 +1310,9 @@ unsigned long AddCertsFromFile(  // NOLINT(runtime/int)
   if (!bio)
     return ERR_get_error();
 
-  while (X509* x509 =
-      PEM_read_bio_X509(bio.get(), nullptr, NoPasswordCallback, nullptr)) {
-    X509_STORE_add_cert(store, x509);
-    X509_free(x509);
+  while (X509Pointer x509 = X509Pointer(
+      PEM_read_bio_X509(bio.get(), nullptr, NoPasswordCallback, nullptr))) {
+    certs.push_back(std::move(x509));
```

Comment:
If `root_cert_store != nullptr` the extra certificates will not be loaded into the `root_cert_store` and will be missing from the "default" secure context. However, the existing logic for `UseExtraCaCerts` and `AddCertsFromFile` only worked when called at initial startup, and that remains true here.